actions/restore.py

```diff
@@ -190,11 +191,20 @@ def probe_forked_etcd():
 def reconfigure_client_advertise():
     ''' Reconfigure the backup to use host network addresses for client advertise
         instead of the assumed localhost addressing '''
-    cmd = "/snap/bin/etcd.etcdctl member list"
-    members = check_output(split(cmd))
-    member_id = members.split(b':')[0].decode('utf-8')
 
-    raw_update = "/snap/bin/etcd.etcdctl member update {0} http://{1}:{2}"
+    loop = 0
+    MAX_WAIT = 10
+
+    while loop < MAX_WAIT:
+        try:
+            cmd = "/snap/bin/etcd.etcdctl member list"
+            members = check_output(split(cmd))
+            member_id = members.split(b',')[0].decode('utf-8')
+            break
+        except:
```

Comment:
instead of catching all exceptions, It would be better to catch only those exceptions that you care about